File: nomad-samples/portal/js/content-groups/content-groups/get-content-groups.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function getContentGroups(AUTH_TOKEN) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
    // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/contentgroup`, {
        method: "GET",
        headers: HEADERS
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Failed to get content group");
}

File: nomad-samples/common/py/movies/content_groups/content_group/get_content_groups.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json
import requests

def get_content_groups(AUTH_TOKEN: str) -> dict:
    if not AUTH_TOKEN:
        raise Exception("Authentication token not found")
  
    API_URL = PORTAL_API_URL + "/contentgroup"
    
    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        RESPONSE = requests.get(API_URL, headers=HEADERS)
        if not RESPONSE.ok:
            raise Exception()

        return json.loads(RESPONSE.text)
   	
    except:
      	api_exception_handler(RESPONSE, "Get content groups failed")